### PR TITLE
math/mpi: add mpi driver in math

### PR DESCRIPTION
**Pull request by ThePassionate**
add interface for MPI(Multiple Precision Integer) registration in /math/mpi
Signed-off-by: makejian <makejian@xiaomi.com>

### CODE REVIEW

File: drivers/math/Make.defs [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 # Include math acceleration support
 
-ifneq ($(CONFIG_MATH_CORDIC)$(CONFIG_MATH_FFT),)
+ifneq ($(CONFIG_MATH_CORDIC)$(CONFIG_MATH_FFT)$(CONFIG_MATH_MPI),)
 CSRCS += math.c
 endif
```
Comment (acassis) [inline drivers/math/Make.defs:26]: Please include also the update in the CMakefile, we need to keep both building systems in sync!

**Review by acassis — CHANGES_REQUESTED** [review]

**Comment by acassis** [comment]
@xiaoxiang781216 is it possible to include an automatic check to verify if a PR modified a Makefile and didn't modify the CMakefile at same level? I think we will need it avoid mistakes.

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 is it possible to include an automatic check to verify if a PR modified a Makefile and didn't modify the CMakefile at same level? I think we will need it avoid mistakes.

@xuxin930 do you have any idea?

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by raiden00pl — APPROVED** [review]
lgtm, except missing cmake support

**Comment by xuxin930** [comment]
> @xuxin930 do you have any idea?

how about simply checking in CI whether both files have been modified? 
whether the content matches or not, I think it still requires human intervention.

**Comment by acassis** [comment]
> > @xuxin930 do you have any idea?
> 
> how about simply checking in CI whether both files have been modified? whether the content matches or not, I think it still requires human intervention.

Exactly, by same level I mean: "Makefile/Make.defs and CMakeList.txt in the same directory need to be modified"

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**